### DISCUSSION TITLE
Plotting slip trace of FCC phase

### DISCUSSION THREAD

**Discussion by abhi12019** [mtex-toolbox/mtex#2063] [Ask Anything]
Hi,

I am working on the problem of plotting slip traces for my EBSD data on steel. I performed an in-situ tensile experiment at 800 degrees C. I got in-situ microstructures showing the slip traces during deformation. After the experiment, the EBSD data was obtained, but during cooling, the austenite phase is transforming to ferrite. Basically, I am using this code for slip trace plotting. 
```matlab
ebsd = EBSD.load(fname,CS,'interface','crc',...
  'convertEuler2SpatialReferenceFrame');
close all
[grains,ebsd.grainId] = calcGrains(ebsd,'angle',15*degree);
% remove small grains
ebsd(grains(grains.diameter<=5)) = [];
% redo grain reconstruction
[grains,ebsd.grainId] = calcGrains(ebsd,'angle',15*degree);

%calculating from basics
sS=slipSystem.fcc(ebsd.CS)
sS=sS.symmetrise;
%rotate slip systems into specimen coordinates
sSlocal=grains.meanOrientation*sS
%compute Schmidt factor
sigma=stressTensor.uniaxial(vector3d.X)
SF=sSlocal.SchmidFactor(sigma);
%take the maximum along the rows
[SFMax, active]=max(SF,[],2);
%plot the maximum schmidt factor
figure
plot(grains, SFMax, 'micronbar', 'off','linewidth',2)
mtexColorbar location southoutside
%take the active slip system and rotate it in specimen coordinates
sSactive=grains.meanOrientation.*sS(active);
hold on
%visualize the trace of the slip plane
quiver(grains, sSactive.trace,'color','b');
% and the slip direction
quiver(grains,sSactive.b,'color','r');
hold off
```
I am getting this results:

band contrast map: This is also showing the direction of slip trace, but the slip trace plot shows different direction 
![band_contrast_maps](https://github.com/mtex-toolbox/mtex/assets/82941844/bb639718-98dc-498e-b00d-d1ba26b9f728)
slip trace results with Schmidt factor
![slip_trace](https://github.com/mtex-toolbox/mtex/assets/82941844/de6769c3-d030-4e67-b24f-8c36665451be)
in-situ images at strain of 0.094
![in-situ_images](https://github.com/mtex-toolbox/mtex/assets/82941844/3933d0b3-18c6-442f-af38-2b2107a8e6b0)

The slip trace is not matching with the in-situ microstructure images. I have defined all the things correctly. I have been reading the manual of slip trace analysis, but I am not able to solve the problem. Please assist me in this problem.

**Comment by kilir** [comment]
Hi,
a few thoughts:
a) are you really sure that the `'convertEuler2SpatialReferenceFrame'` which is simply a hard-coded rotation around mtex-xvector by 180 degree is correct for your SEM/EBSD setup. This choice of rotation is somewhat somethign that seems to apply for some setups, but there's no guarantee that it is correct. 
b) are you really sure that the x-direction in mtex is also the one you pulled on? Could you see some sample feature which you can relate in both your tensile stage (x direction) to check your specimen direction (mtex x direction)?

Cheers,

Rüdiger

**Reply by abhi12019** [reply]
Hi Rudiger,

Thank you for your comments. These are my replies for your comments. I have read the manual for EBSD data importing, they have also mentioned the same default setting for importing the EBSD data from crc file.
(a). I have used the default setting using import wizard of MTEX. This is the screenshot of selecting the reference system. 
![image](https://github.com/mtex-toolbox/mtex/assets/82941844/c2f5ca93-6c7d-43da-9570-2bd89b130a80)
I am using this convention to get the same orientation as I am getting in the Aztec software EBSD results.
(b). I have similar sample features as in the in-situ images with electron EBSD images. Based on this, I think the tensile direction is X-direction. These are the electron images which shows the twin boundary at the right top marked with the arrow.
![image](https://github.com/mtex-toolbox/mtex/assets/82941844/6da725e6-fbf3-4407-b4db-0fba5c99af0b)
This is the electron image with band contrast mapped on this images.
![image](https://github.com/mtex-toolbox/mtex/assets/82941844/74b5ba78-d778-4053-86cf-2f6206611b6e)
This is one of the in-situ images to see the twin boundary and the area where the EBSD scanning is done. The deformation is applied along the X direction shown. 
![image](https://github.com/mtex-toolbox/mtex/assets/82941844/de501f40-1ce7-4dd2-ad0c-ba847e124f26)

I have tried to change the tensile direction, but the slip trace is not matching with the in-situ images. I am not able to figure out this problem. Please help me

**Reply by kilir** [reply]
>I am using this convention to get the same orientation as I am getting in the Aztec software EBSD results.

Beware that the tiny x,y label in the Aztec pf may point to different directions than x,y of the map, it is consistent once loaded into Aztec crystal, but I guess you triple checked that.

Another thought, maybe the difference between the assumtion of homogeneous state of stress and reality?

**Comment by ZGDongYuan** [comment]
I see the [SFMax, active]=max(SF,[],2) code used, which shows the slippage surface with the greatest Schmitt factor, so the slippage trace is also the surface with the greatest Schmitt factor shown. However, the slip trace may not be on the slip surface with the greatest Schmitt factor. You can try using the OIM calibration to superimpose the traces of all the slip surfaces.
